tests/files/JCEProviderInfo.java

```diff
@@ -0,0 +1,42 @@
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.Provider;
+import java.security.Provider.Service;
+import java.security.Security;
+
+public class JCEProviderInfo
+{
+    public static void main(String[] args) throws NoSuchAlgorithmException, NoSuchProviderException
+    {
+        System.out.printf("JCE Provider Info: %s %s/%s on %s %s%n", System.getProperty("java.vm.name"),
```

Comment:
I would print only the actual providers to `System.out` and all of the debug/information stuff to `System.err`, to make it better compatible with other tools and not confuse them

Reply:
```suggestion
        System.err.printf("JCE Provider Info: %s %s/%s on %s %s%n", System.getProperty("java.vm.name"),
```

Reply:
:heavy_check_mark: